Clean code related to savedObjectClientContrat that wasn't being used

jesuswr opened the pull request:
## Summary

Resolves: https://github.com/elastic/kibana/issues/197216

Removed all code that wasn't being used in https://github.com/elastic/kibana/tree/main/src/plugins/saved_objects/public.
Didn't remove the actual client since it's still being used here: https://github.com/elastic/kibana/blob/dbab2214e9451d3a262007660cc0b7cdcea5307c/src/plugins/home/public/application/kibana_services.ts#L46-L48

Checked everything with `node scripts/type_check`

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_node:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

dmlemeshko committed:
[ftr] update docs with correct auth arguments (#200532)

## Summary

Update FTR docs both in `x-pack/test_serverless` and
`x-pack/test/api_integration/deployment_agnostic/` paths to show
relevant example for authentication with Cookie header.

rmyz committed:
[APM] Migrate `/service_maps` to deployment agnostic test (#199984)

## Summary

Closes https://github.com/elastic/kibana/issues/198983
Part of https://github.com/elastic/kibana/issues/193245

This PR contains the changes to migrate `service_maps` test folder to
Deployment-agnostic testing strategy.

### How to test

- Serverless

```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts --grep="APM"
```

It's recommended to be run against
[MKI](https://github.com/crespocarlos/kibana/blob/main/x-pack/test_serverless/README.md#run-tests-on-mki)

- Stateful
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts --grep="APM"
```

## Checks

- [ ] (OPTIONAL, only if a test has been unskipped) Run flaky test suite
- [x] local run for serverless
- [x] local run for stateful
- [x] MKI run for serverless

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign test files to ml-ui team (#200212)

## Summary

Assign test files to ml-ui team

Contributes to: #192979

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Graham Hudgins <graham@elastic.co>
Co-authored-by: Irene Blanco <irene.blanco@elastic.co>
Co-authored-by: Pablo Machado <pablo.nevesmachado@elastic.co>
Co-authored-by: Robert Oskamp <robert.oskamp@elastic.co>
Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>
Co-authored-by: Maxim Palenov <maxim.palenov@elastic.co>
Co-authored-by: Dmitrii Shevchenko <dmitrii.shevchenko@elastic.co>
Co-authored-by: Nicolas Chaulet <nicolas.chaulet@elastic.co>
Co-authored-by: Ania Kowalska <63072419+akowalska622@users.noreply.github.com>
Co-authored-by: Milosz Marcinkowski <38698566+miloszmarcinkowski@users.noreply.github.com>
Co-authored-by: Sergi Romeu <sergi.romeu@elastic.co>
Co-authored-by: Nathan Reese <reese.nathan@elastic.co>
Co-authored-by: Lisa Cawley <lcawley@elastic.co>
Co-authored-by: Viduni Wickramarachchi <viduni.wickramarachchi@elastic.co>
Co-authored-by: wajihaparvez <wajiha.parvez@elastic.co>

kowalczyk-krzysztof committed:
[Papercut] Fix KQL parsing error text styling (#199985)

## Summary

This PR fixes styling for `KQL parsing error` text on various elements
on `Maps` and `Dashboard` to have the invalid syntax and ASCII arrow on
new line.

Closes: #49377


![dashboard1](https://github.com/user-attachments/assets/c607c0e0-24d8-4bd9-8106-d5c94fe1197d)

![dashboard2](https://github.com/user-attachments/assets/cb4b58af-6d8b-4609-9a4e-b948b1ec9340)

![dashboard3](https://github.com/user-attachments/assets/5a6c883e-76dc-4f9d-8db6-94c3e0bc359a)

![maps](https://github.com/user-attachments/assets/83d897a4-6856-4c47-8b14-18a1fff1de9d)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

tkajtoch committed:
Add basic support for experimental theme switching (#199748)

## Summary

This PR adds basic support for theme switching to test experimental
themes internally. It defines a new `theme:name` setting which is
read-only (and thus hidden) by default, and needs
`uiSettings.experimental.themeSwitcherEnabled: true` config setting to
become visible. The implementation and the way the theme switcher
feature is enabled will likely change in the upcoming weeks when we iron
out the details, but the way it works right now is sufficient for
testing and that's what the immediate goal is.

Please note this PR includes mostly setup work, and no additional themes
have been added here. To make reviewing slightly easier, these will be
added separately.

The feature and settings should be undocumented for the time being and
disabled by default.

---

As you might notice, there's a new setting added despite already having
`theme:version`. We decided to introduce a new setting instead of
resurrecting an existing one for better naming and flexibility and to
reduce the risk of breaking things that might still use the old setting
value (hardcoded to `v8` at the moment). The current plan is to remove
`theme:version` in 9.0.

The theme_loader got refactored to only bundle active themes coming from
`KBN_OPTIMIZER_THEMES` (defaulting to `v8light` and `v8dark`) instead of
`ALL_THEMES` that would significantly increase bundle sizes even when
the experimental themes were not enabled. Considering there's a SASS to
CSS-in-JS migration happening right now, we don't need to worry about
Kibana bundles growing in size when a new theme is officially added.

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)
- [ ] This will appear in the **Release Notes** and follow the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Clint Andrew Hall <clint.hall@elastic.co>
Co-authored-by: Rudolf Meijering <skaapgif@gmail.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Tiago Costa <tiago.costa@elastic.co>

wayneseymour committed:
[Ownership] Assign test files to obs-ux-management-team (#200176)

## Summary

Assign test files to obs-ux-management-team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign test files to obs-ux-infra_services-team team (#200156)

## Summary

Assign test files to obs-ux-infra_services-team team

Contributes to: #192979

wayneseymour committed:
[Ownership] Assign test files to obs-ux-logs team (#200157)

## Summary

Assign test files to obs-ux-logs team

Contributes to: #192979

miloszmarcinkowski committed:
Migrate /test/apm_api_integration/tests/span_links to be deployment-agnostic API tests (#200140)

closes #198990
part of https://github.com/elastic/kibana/issues/193245

### How to test

- Serverless

```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts --grep="APM"
```

- Stateful
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts --grep="APM"
```

-
[MKI](https://github.com/crespocarlos/kibana/blob/main/x-pack/test_serverless/README.md#run-tests-on-mki)

### Checklist
- [x] (OPTIONAL, only if a test has been unskipped) Run flaky test suite
- [x] serverless
- [x] stateful
- [x] MKI

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kowalczyk-krzysztof committed:
[Lens] Prefer pie over donut. Keep donut if donut already (#199735)

## Summary

This PR makes it so `pie` is used over `donut` as default suggestion. If
something is a `donut` already, adding another ring won't switch to
`pie`.

Closes: #101289

wayneseymour committed:
[Ownership] Assign sample data to core team (#200142)

## Summary

Assign test files to platform docs team

Contributes to: #192979

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign test files to ops team (#200217)

## Summary

Assign test files to ops team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

janmonschke committed:
[SecuritySolution][Timeline] Remove timeline.isLoading (#198616)

## Summary

Trying to answer the question of: "Do we still need
`timeline.isLoading`?"

`timeline.isSaving` should be the only indicator for "loading" states of
timeline itself. All other pieces of state that are associated with
timeline that could have a loading state, have their own loading
indicators (e.g. data providers, alert list etc).

Therefore, this PR removes all references to `timeline.isLoading` and
parts of the UI that depended on it.

Places that `timeline.isLoading` was used
([context](https://github.com/elastic/kibana/pull/38185)):
- Blocking drag/drop on data providers
- This is not necessary anymore. Drag/drop works while the underlying
query is being executed.
- Showing a loading state for the alerts table & data provider changes
- Both components have their own loading state, so no extra loading
state is necessary

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

jennypavlova committed:
[Infra] Change order of the errors shown on Infra pages (#200531)

Closes #200190 

## Summary

This PR fixes the issue with the errors shown on the metrics explorer
page - when the metrics indices can't be fetched we should show the
error returned and if there is no fetch error and a remote cluster
configured but no connection is possible we should show the remote
cluster error:

- Example with 504 error 

![image](https://github.com/user-attachments/assets/65cd8226-8c81-4c64-b043-c9db5a93d3e0)

- Example with remote cluster error

![image](https://github.com/user-attachments/assets/e024a3f8-76e0-4ad7-8aa6-e35ad5c1112a)


## Testing
Couldn't find a way to reproduce this so I "faked" the API response to
be an error and checked several cases
- API returns an error, we should show the error:


https://github.com/user-attachments/assets/c1086b22-1ff5-4333-97a5-b3d1dca16afe


- API doesn't return an error but the remote cluster connection wasn't
possible, we should show the remote cluster error:


https://github.com/user-attachments/assets/151b3ae4-5ca1-4d54-bd58-2729db202cdb


- If no remote cluster is used/or a remote cluster is connected and the
API response is not returning an error the page should load correctly:


https://github.com/user-attachments/assets/f9ef1066-3dfd-4957-8b46-878bf58d2f1c

wayneseymour committed:
[Ownership] Assign test files to apm teams (#200182)

## Summary

Assign test files to apm teams

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign test files to stack-monitoring team (#200178)

## Summary

Assign test files to stack-monitoring team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign test files to observability-ui team (#200185)

## Summary

Assign test files to observability-ui team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

sabarasaba committed:
[Console] Avoid console url redirection (#199966)

szwarckonrad committed:
[EDR Workflows] Add Signer option to Mac trusted apps (#197821)

This PR adds a Signer condition for trusted apps on macOS. Previously,
users could only build conditions using hash, path, and signer options
on Windows. With these changes, macOS also supports the Signer option,
leaving only Linux limited to Path and Hash options.


https://github.com/user-attachments/assets/ea8fb734-7884-451d-8873-e3a29861876b

jesuswr committed:
remove all code related to savedObjectClientContrat that wasn't being used

gsoldevila committed:
Kibana Sustainable Architecture: Expose `StatusResponse` in core-status-common (#200524)

## Summary

- Exposes the interfaces that define the format of the response of the
`/api/status` endpoint.
- Moves them from `@kbn/core-status-common-internal` to
`@kbn/core-status-common`.
- Removes the former package, as it no longer contains anything.
- Fixes some of the illegal dependencies uncovered by
https://github.com/elastic/kibana/pull/199630.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[EuiInMemoryTable] Persist table rows per page and sort (#198297)

kibanamachine committed:
Unauthorized route migration for routes owned by security-defend-workflows (#198375)

### Authz API migration for unauthorized routes

Fix unauthorized routes

david-luna committed:
feat: update OTEL Node.js metrics dashboard (#199353)

## Summary

Update the current metrics dashboard for Node.js services that are
instrumented using OTEL SDK or Elastic's distribution.

The current dashboard had some issues:
- it was using a view that is not available in latest deployments.
- it was showing labels instead of OTEL native fields

New dashboard is fixing this:
- using `metrics-*` indices
- using OTEL native fields
- and also adding new panels for `nodejs.*` metrics
([ref](https://github.com/open-telemetry/semantic-conventions/blob/e0e93bad394a15161af7942d65d197defac7b88b/model/nodejs/metrics.yaml))

legrego committed:
Dependency ownership for AI Infra Team (#200238)

## Summary

This updates our `renovate.json` configuration to mark the AI Infra team
as owners of their set of dependencies.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

TinaHeiligers committed:
Improve logging pattern schema validation (#200317)

## Summary

This makes pattern schema validation stricter by avoiding unrestricted
matching, enforces realistic timezone constraints, and sets an upper
bound to the date format string.

| Characteristic | Before | After |
----|----|----|
| Optional Groups | Capturing groups for optional sections. |
Non-capturing groups for optional sections.|
|Timezone Validation | Allows any characters except }. | Restricts to
A-Za-z/_+-.|
|Clarity | More complex and verbose. | Cleaner, more restrictive, and
simpler.|
|Performance | Captures unnecessary groups. | Limits captures to needed
named groups.|

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/251970)
* Commit: e2393e5304938bb7c055e228f841776a8aa78c3a

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/251970#01933fb8-a566-4416-97b0-b0fe5eadcf6e)

cc @jesuswr

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/251970","number":251970,"commit":"e2393e5304938bb7c055e228f841776a8aa78c3a"}],"number":251970}
buildkite-pr-comment-->

sphilipse committed:
Search fix documents table a11y (#200232)

## Summary

This improves the accessibility of the documents table by:
1. Adding header rows with titles
2. Adding an accessible table description
3. Adding an accessible popover to the field type icon

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

mikecote committed:
Update doc on ruleTaskTimeout to mention requests to Elasticsearch also receive the same timeout, up to 5m (#200172)

Updating the docs to mention ruleTaskTimeout also affects requests made
to Elasticsearch

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

abhishekbhatia1710 committed:
[Entity Analytics] API changes for right placement of deleting the old component template (#199734)

## Summary

- Delete the old component template after the index template has
referenced to the new component template
- Test cases for the same flow


```JSON
# Let's assume this is 8.15.3
# Create the component template when Risk Score engine is initialised
# Create the index template which references the created component template

PUT /_component_template/.risk-score-mappings
{
  "template": {
    "settings": {
      "number_of_shards": 1
    },
    "mappings": {
      "properties": {
        "timestamp": {
          "type": "date"
        },
        "user": {
          "properties": {
            "id": {
              "type": "keyword"
            },
            "name": {
              "type": "text"
            }
          }
        }
      }
    }
  },
  "version": 1
}

PUT /_index_template/.risk-score.risk-score-default-index-template
{
  "index_patterns": [".risk-score.risk-score-default-index-template"],
  "template": {
    "settings": {
      "number_of_replicas": 1
    }
  },
  "composed_of": [".risk-score-mappings"],
  "priority": 100,
  "version": 1,
  "_meta": {
    "description": "Index template for indices with the pattern my_index-*"
  }
}

# The deployment is updated to 8.16
# User tries to enable the Entity store which init's the Risk Score engine (again!!) 
# Fails, but creates the component template and cannot update the index template to reference the new component template due to the error

PUT /_component_template/.risk-score-mappings-default
{
  "template": {
    "settings": {
      "number_of_shards": 1
    },
    "mappings": {
      "properties": {
        "timestamp": {
          "type": "date"
        },
        "user": {
          "properties": {
            "id": {
              "type": "keyword"
            },
            "name": {
              "type": "text"
            }
          }
        }
      }
    }
  },
  "version": 1
}

GET /_component_template?filter_path=component_templates.name&name=.risk-score-mappings*


DELETE /_component_template/.risk-score-mappings

# Fails 

# changed flow
PUT /_index_template/.risk-score.risk-score-default-index-template
{
  "index_patterns": [".risk-score.risk-score-default-index-template"],
  "template": {
    "settings": {
      "number_of_replicas": 1
    }
  },
  "composed_of": [".risk-score-mappings-default"],
  "priority": 100,
  "version": 1,
  "_meta": {
    "description": "Index template for indices with the pattern my_index-*"
  }
}

DELETE /_component_template/.risk-score-mappings

# Succeeds

#########
```


### Checklist

Delete any items that are not applicable to this PR.

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed



### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#_add_your_labels)
- [ ] This will appear in the **Release Notes** and follow the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

dej611 committed:
[Visualize] Fix embeddable panel title behavior (#200548)

## Summary

This PR adds a fix for a regression bug introduced with the new
embeddable refactor in 8.16 .
I've added an extra 8.16 FTR test to ensure it works.


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

nchaulet committed:
[Fleet] Fix agent policy namespace validation (#200258)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/252005)
* Commit: 0584227c0b58a38162155c3f05abc039eb852967

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/252005#01933fe7-2522-4349-9103-52e21d4c7bd2)

### History
* :broken_heart: [Build #251970](https://buildkite.com/elastic/kibana-pull-request/builds/251970) failed e2393e5304938bb7c055e228f841776a8aa78c3a

cc @jesuswr

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/252005","number":252005,"commit":"0584227c0b58a38162155c3f05abc039eb852967"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/251970","number":251970,"commit":"e2393e5304938bb7c055e228f841776a8aa78c3a"}],"number":252005}
buildkite-pr-comment-->

stratoula committed:
[ES|QL] Don't suggest unsupported fields (#200544)

## Summary

Closes https://github.com/elastic/kibana/issues/189666

Specifically:

- we do not suggest unsupported fields
- we add a warning if the field is used in the query

### How to test

1. Create an index with unsupported fields. I used this

```
PUT my-index-test-unsupported
{
  "mappings" : {
    "properties" : {
      "my_histogram" : {
        "type" : "histogram"
      },
      "my_text" : {
        "type" : "keyword"
      }
    }
  }
}

PUT my-index-test-unsupported/_doc/1
{
  "my_text" : "histogram_1",
  "my_histogram" : {
      "values" : [0.1, 0.2, 0.3, 0.4, 0.5], 
      "counts" : [3, 7, 23, 12, 6] 
   }
}

PUT my-index-test-unsupported/_doc/2
{
  "my_text" : "histogram_2",
  "my_histogram" : {
      "values" : [0.1, 0.25, 0.35, 0.4, 0.45, 0.5], 
      "counts" : [8, 17, 8, 7, 6, 2] 
   }
}
```

2. Go to the editor and try the autocomplete, it should not suggest the
histogram field

<img width="998" alt="image"
src="https://github.com/user-attachments/assets/2a9b4594-1f8f-4e7a-9ea3-d9086c770178">

3. Use a query like `FROM my-index-test-unsupported | KEEP my_histogram
`. You should see a warning

<img width="991" alt="image"
src="https://github.com/user-attachments/assets/1bf747da-191f-46e2-b1b8-e5966653f405">


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

SiddharthMantri committed:
Fix issue with duplicate references in error object when copying saved objects to space (#200053)

Closes https://github.com/elastic/kibana/issues/158027

## Summary

Simply dedupes references to objects if they are part of the
missing_references in the copy saved objects to SO endpoint

### Notes
- Update forEach over SOs to a regular for loop since we had a couple of
early exit scenarios
- Checks against the set for references already added to the missing
list and adds only if not present

------

**Old response: Note the duplicate references**

<img width="400" alt="Screenshot 2024-11-14 at 01 52 54"
src="https://github.com/user-attachments/assets/67078080-e39d-43b2-bf7c-7abb76866fa4">


**New response**

<img width="800" alt="Screenshot 2024-11-14 at 01 50 41"
src="https://github.com/user-attachments/assets/776db189-af8c-4522-bb03-f8efbb7cdcd9">


### Release note
Dedupe results from copy saved objects to spaces API when object
contains references to other objects.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

smith committed:
Respect system dark mode in ESQL editor (#200233)

When `uiSettings.overrides.theme:darkMode: true` is set, the ESQL editor
uses dark mode.

When `uiSettings.overrides.theme:darkMode: system` is set, the ESQL
editor uses light mode, while the rest of the UI uses dark mode.

Update the ESQL theme creation to create both light and dark variations
of the theme at startup and use the theme in React component to
determine which ESQL theme to use at runtime.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

saarikabhasi committed:
[Search][Onboarding] Enable or disable index actions based on user privileges (#197869)

## Summary
Disable or show index actions in the search indices index details page
based on user privileges. Stack should work as it is, this change is
relevant to Serverless search index details page.

based on user privilege below buttons are disabled:
1.  delete index in index action context menu
2. delete documents in data tab
3. add field in mappings tab
4. edit index settings in setting tab


### Screen recording 


https://github.com/user-attachments/assets/047f0d9f-b6c3-4665-9cff-a7a4d978b03b

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

wayneseymour committed:
[Ownership] Assign maps test files to presentation team (#200214)

## Summary

 Assign maps test files to presentation team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Nick Peihl <nickpeihl@gmail.com>

e40pud committed:
[Security Solution] [Bug] Index Values are not available in dropdown under New Index Enter for Knowledge Base. (#199610) (#199990)

## Summary

This PR fixes the issue described here
https://github.com/elastic/kibana/issues/199610#issuecomment-2473393109

In short, we do not show all indices with `semantic_text` fields. We
only show those which have `semantic_text` fields named `content`.

### Testing notes

Add the index with the `semantic_text` field by running this command in
dev tools:

```
PUT test_index
{
  "mappings": {
  "properties": {
    "attachment": {
      "properties": {
        "content": {
          "type": "text",
          "fields": {
            "keyword": {
              "type": "keyword",
              "ignore_above": 256
            }
          }
        },
        "content_length": {
          "type": "long"
        },
        "content_type": {
          "type": "text",
          "fields": {
            "keyword": {
              "type": "keyword",
              "ignore_above": 256
            }
          }
        },
        "format": {
          "type": "text",
          "fields": {
            "keyword": {
              "type": "keyword",
              "ignore_above": 256
            }
          }
        },
        "language": {
          "type": "text",
          "fields": {
            "keyword": {
              "type": "keyword",
              "ignore_above": 256
            }
          }
        }
      }
    },
    "content2": {
      "type": "semantic_text",
      "inference_id": "elastic-security-ai-assistant-elser2"
    }
    }
}
}
```

You should see the `test_index` index in the `Knowledge Base > New >
Index > Index (dropdown)`.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Steph Milovic <stephanie.milovic@elastic.co>

e40pud committed:
[Security GenAI] Fetching Assistant Knowledge Base fails when current user's username contains a : character (#11159) (#200131)

## Summary

Original bug: [internal
link](https://github.com/elastic/security-team/issues/11159)

**This PR fixes the next bug**:
When the user is logged in with a username that contains a `:`
character, fetching Knowlege Base entries fails with an error. This is
preventing customers from viewing their created KB entries. This problem
affects ECE customers using the SSO login option.

There were a similar bugfix which inspired this one
https://github.com/elastic/kibana/pull/181709

There is no easy way to reproduce this but you can try and change the
line in question so that the faulty username is used instead of the one
passed in.

@MadameSheema Do you know a way to login locally with the username that
contains a `:` character? As mentioned above this situation is possible
with the ECE customers using SSO login.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jennypavlova committed:
[Inventory] Remove open in Discover button (#200574)

Closes #199964 

This PR removes the `open in discover` button from the inventory search
bar section.

| Before | After |
| ------- | ----- |
| <img width="1911" alt="image"
src="https://github.com/user-attachments/assets/627ebfa0-77c9-4ff1-8a38-3623bf1b7e32">
| <img width="1915" alt="image"
src="https://github.com/user-attachments/assets/2dace203-7203-4bda-8cee-0186e022e373">
|

wayneseymour committed:
[Ownership] Assign test files to platform security team (#199795)

## Summary

Assign test files to platform security team

Contributes to: #192979

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Jeramy Soucy <jeramy.soucy@elastic.co>

nkhristinin committed:
Update roles in serverless for value lists (#200128)

Update serverless role for value lists related to this
[PR](https://github.com/elastic/elasticsearch-controller/pull/771)

Commit:
[Obs AI Assistant] Support querying `semantic_text` fields in search connectors (#200184)

This adds support for querying search connectors that have
`semantic_text` fields.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kyracho committed:
[Lens] rewrite BucketNestingEditor tests to use react testing library (#199888)

## Summary

Hi! This PR rewrites the `BucketNestingEditor` tests to use the react
testing library.

Fixes #199839 

<img width="1063" alt="Screenshot 2024-11-12 at 3 35 06 PM"
src="https://github.com/user-attachments/assets/95036fe0-46e8-4bf2-9d77-31d225d38ed1">

### Checklist

- [n/a] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed

---------

Co-authored-by: Marta Bondyra <4283304+mbondyra@users.noreply.github.com>

Ruby-rc committed:
[Console]: incorrect position of console tour #198389 (#198636)

crespocarlos committed:
[APM] Migrate time_range_metadata test to deployment agnostic (#200146)

## Summary

Closes [#198994](https://github.com/elastic/kibana/issues/198994)
Part of https://github.com/elastic/kibana/issues/193245

This PR contains the changes to migrate `time_range_metadata` test
folder to Deployment-agnostic testing strategy.

### How to test

- Serverless

```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts --grep="APM"
```

It's recommended to be run against
[MKI](https://github.com/crespocarlos/kibana/blob/main/x-pack/test_serverless/README.md#run-tests-on-mki)

- Stateful
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts --grep="APM"
```

- [ ] ~(OPTIONAL, only if a test has been unskipped) Run flaky test
suite~
- [x] local run for serverless
- [x] local run for stateful
- [x] MKI run for serverless

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

miloszmarcinkowski committed:
Migrate `/test/apm_api_integration/tests/suggestions` to be deployment agnostic api tests (#200556)

closes #198992
closes #198993
part of https://github.com/elastic/kibana/issues/193245

### How to test

- Serverless

```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/serverless/oblt.serverless.config.ts --grep="APM"
```

- Stateful
```
node scripts/functional_tests_server --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts
node scripts/functional_test_runner --config x-pack/test/api_integration/deployment_agnostic/configs/stateful/oblt.stateful.config.ts --grep="APM"
```

-
[MKI](https://github.com/crespocarlos/kibana/blob/main/x-pack/test_serverless/README.md#run-tests-on-mki)

### Checklist
- [x] (OPTIONAL, only if a test has been unskipped) Run flaky test suite
- [x] serverless
- [x] stateful
- [x] MKI

---------

Co-authored-by: Sergi Romeu <sergi.romeu@elastic.co>

wayneseymour committed:
[Ownership] Assign test files to presentation team (#200209)

## Summary

Assign test files to presentation team

Contributes to: #192979

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

TinaHeiligers reviewed (commented):
We'll need to remove the unused translation strings from our i18n `.json` [files](https://github.com/elastic/kibana/tree/main/x-pack/plugins/translations/translations).
Running `node scripts/i18n_check --fix` will do that automatically. The [README](https://github.com/elastic/kibana/blob/main/src/dev/i18n_tools/README.md) explains what the `--fix` does

jesuswr committed:
fix translation files with script

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

elasticmachine commented:
Pinging @elastic/obs-ai-assistant (Team:Obs AI Assistant)

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/252111)
* Commit: 5c485fa728713f6e8fe03193c0bb6aa08be49f14
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-200577/5c485fa728713f6e8fe03193c0bb6aa08be49f14)

### Failed CI Steps
* [Serverless Rule Management - Prebuilt Rules - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/252111#01934074-2f09-45ae-86d4-6732c2381a4e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `advancedSettings` | 110 | 111 | +1 |
| `aiops` | 627 | 625 | -2 |
| `alerting` | 239 | 248 | +9 |
| `banners` | 33 | 34 | +1 |
| `canvas` | 1291 | 1292 | +1 |
| `cases` | 820 | 821 | +1 |
| `cloudDataMigration` | 22 | 23 | +1 |
| `cloudLinks` | 78 | 79 | +1 |
| `cloudSecurityPosture` | 662 | 663 | +1 |
| `console` | 238 | 236 | -2 |
| `controls` | 357 | 364 | +7 |
| `core` | 447 | 445 | -2 |
| `crossClusterReplication` | 133 | 134 | +1 |
| `dashboard` | 679 | 687 | +8 |
| `dashboardEnhanced` | 96 | 97 | +1 |
| `data` | 517 | 525 | +8 |
| `dataQuality` | 89 | 90 | +1 |
| `dataUsage` | 133 | 134 | +1 |
| `dataViewEditor` | 86 | 87 | +1 |
| `dataViewFieldEditor` | 162 | 163 | +1 |
| `dataViewManagement` | 241 | 249 | +8 |
| `dataVisualizer` | 735 | 733 | -2 |
| `devTools` | 43 | 44 | +1 |
| `discover` | 977 | 980 | +3 |
| `embeddable` | 143 | 144 | +1 |
| `enterpriseSearch` | 2321 | 2319 | -2 |
| `esql` | 92 | 99 | +7 |
| `eventAnnotationListing` | 585 | 593 | +8 |
| `exploratoryView` | 182 | 183 | +1 |
| `expressionError` | 40 | 26 | -14 |
| `expressionGauge` | 107 | 108 | +1 |
| `expressionHeatmap` | 182 | 183 | +1 |
| `expressionImage` | 26 | 12 | -14 |
| `expressionLegacyMetricVis` | 47 | 48 | +1 |
| `expressionMetric` | 24 | 10 | -14 |
| `expressionMetricVis` | 115 | 116 | +1 |
| `expressionPartitionVis` | 196 | 197 | +1 |
| `expressionRepeatImage` | 24 | 10 | -14 |
| `expressionRevealImage` | 27 | 13 | -14 |
| `expressionShape` | 70 | 56 | -14 |
| `expressionTagcloud` | 170 | 171 | +1 |
| `expressionXY` | 257 | 258 | +1 |
| `filesManagement` | 191 | 199 | +8 |
| `fleet` | 1238 | 1239 | +1 |
| `globalSearchBar` | 48 | 49 | +1 |
| `graph` | 303 | 311 | +8 |
| `grokdebugger` | 41 | 42 | +1 |
| `guidedOnboarding` | 42 | 43 | +1 |
| `home` | 267 | 268 | +1 |
| `imageEmbeddable` | 141 | 142 | +1 |
| `indexLifecycleManagement` | 261 | 263 | +2 |
| `indexManagement` | 696 | 698 | +2 |
| `infra` | 1526 | 1529 | +3 |
| `ingestPipelines` | 358 | 357 | -1 |
| `inputControlVis` | 94 | 95 | +1 |
| `inspector` | 78 | 79 | +1 |
| `interactiveSetup` | 58 | 59 | +1 |
| `inventory` | 276 | 278 | +2 |
| `investigateApp` | 583 | 571 | -12 |
| `kibanaOverview` | 137 | 138 | +1 |
| `kibanaReact` | 214 | 212 | -2 |
| `kibanaUtils` | 176 | 164 | -12 |
| `lens` | 1471 | 1479 | +8 |
| `licenseManagement` | 125 | 126 | +1 |
| `licensing` | 29 | 30 | +1 |
| `links` | 131 | 132 | +1 |
| `logsShared` | 701 | 703 | +2 |
| `logstash` | 75 | 76 | +1 |
| `management` | 130 | 131 | +1 |
| `maps` | 1244 | 1258 | +14 |
| `metricsDataAccess` | 293 | 294 | +1 |
| `ml` | 2139 | 2140 | +1 |
| `mockIdpPlugin` | 38 | 36 | -2 |
| `monitoring` | 494 | 495 | +1 |
| `navigation` | 77 | 78 | +1 |
| `newsfeed` | 27 | 28 | +1 |
| `observability` | 1069 | 1074 | +5 |
| `observabilityAIAssistantApp` | 385 | 373 | -12 |
| `observabilityLogsExplorer` | 208 | 211 | +3 |
| `observabilityOnboarding` | 255 | 253 | -2 |
| `osquery` | 345 | 346 | +1 |
| `painlessLab` | 45 | 46 | +1 |
| `presentationPanel` | 105 | 112 | +7 |
| `profiling` | 305 | 306 | +1 |
| `remoteClusters` | 125 | 126 | +1 |
| `reporting` | 152 | 153 | +1 |
| `rollup` | 146 | 147 | +1 |
| `runtimeFields` | 34 | 35 | +1 |
| `savedObjects` | 42 | 37 | -5 |
| `savedObjectsFinder` | 8 | 14 | +6 |
| `savedObjectsManagement` | 115 | 123 | +8 |
| `savedObjectsTagging` | 106 | 107 | +1 |
| `searchAssistant` | 253 | 241 | -12 |
| `searchHomepage` | 152 | 153 | +1 |
| `searchIndices` | 235 | 236 | +1 |
| `searchInferenceEndpoints` | 103 | 104 | +1 |
| `searchNotebooks` | 55 | 41 | -14 |
| `searchPlayground` | 269 | 270 | +1 |
| `searchprofiler` | 78 | 79 | +1 |
| `security` | 527 | 528 | +1 |
| `securitySolution` | 6202 | 6212 | +10 |
| `serverless` | 78 | 79 | +1 |
| `serverlessSearch` | 290 | 291 | +1 |
| `share` | 93 | 91 | -2 |
| `slo` | 869 | 867 | -2 |
| `snapshotRestore` | 211 | 212 | +1 |
| `spaces` | 338 | 339 | +1 |
| `stackAlerts` | 162 | 169 | +7 |
| `synthetics` | 1039 | 1037 | -2 |
| `telemetry` | 54 | 55 | +1 |
| `timelines` | 140 | 141 | +1 |
| `transform` | 487 | 488 | +1 |
| `triggersActionsUi` | 853 | 862 | +9 |
| `uiActions` | 40 | 41 | +1 |
| `unifiedSearch` | 376 | 377 | +1 |
| `upgradeAssistant` | 178 | 179 | +1 |
| `uptime` | 590 | 588 | -2 |
| `ux` | 201 | 199 | -2 |
| `visDefaultEditor` | 206 | 207 | +1 |
| `visTypeMarkdown` | 32 | 33 | +1 |
| `visTypeTable` | 50 | 51 | +1 |
| `visTypeTimelion` | 66 | 67 | +1 |
| `visTypeTimeseries` | 462 | 463 | +1 |
| `visTypeVega` | 280 | 281 | +1 |
| `visTypeVislib` | 126 | 127 | +1 |
| `visualizations` | 478 | 492 | +14 |
| `watcher` | 189 | 190 | +1 |
| total |  |  | +73 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `@kbn/core-status-common` | 2 | 22 | +20 |
| `@kbn/core-status-common-internal` | 20 | - | -20 |
| `@kbn/core-ui-settings-common` | 8 | 20 | +12 |
| `@kbn/index-management-shared-types` | 124 | 126 | +2 |
| `@kbn/management-settings-ids` | 138 | 139 | +1 |
| `@kbn/monaco` | 123 | 124 | +1 |
| `@kbn/react-hooks` | 7 | 8 | +1 |
| `@kbn/react-kibana-context-common` | 2 | 8 | +6 |
| `@kbn/react-kibana-context-theme` | 2 | 3 | +1 |
| `@kbn/shared-ux-table-persist` | 2 | 16 | +14 |
| `@kbn/ui-shared-deps-src` | 50 | 51 | +1 |
| `indexManagement` | 238 | 239 | +1 |
| `savedObjects` | 109 | 96 | -13 |
| `securitySolution` | 120 | 119 | -1 |
| total |  |  | +26 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `aiops` | 587.1KB | 586.1KB | -1011.0B |
| `alerting` | 93.8KB | 94.0KB | +171.0B |
| `apm` | 3.4MB | 3.4MB | +3.1KB |
| `canvas` | 1.1MB | 1.1MB | +190.0B |
| `cases` | 491.8KB | 491.9KB | +23.0B |
| `cloudDataMigration` | 5.9KB | 6.0KB | +162.0B |
| `cloudSecurityPosture` | 509.4KB | 509.6KB | +167.0B |
| `console` | 205.6KB | 205.9KB | +313.0B |
| `controls` | 458.6KB | 458.9KB | +311.0B |
| `crossClusterReplication` | 146.2KB | 146.5KB | +249.0B |
| `dashboard` | 645.3KB | 647.0KB | +1.7KB |
| `data` | 48.1KB | 52.4KB | +4.3KB |
| `dataQuality` | 28.2KB | 28.3KB | +168.0B |
| `dataUsage` | 237.9KB | 238.1KB | +191.0B |
| `dataViewManagement` | 137.1KB | 139.8KB | +2.7KB |
| `dataVisualizer` | 615.1KB | 614.0KB | -1.1KB |
| `devTools` | 7.7KB | 7.9KB | +162.0B |
| `discover` | 809.9KB | 810.1KB | +186.0B |
| `embeddable` | 2.8KB | 2.8KB | +24.0B |
| `enterpriseSearch` | 2.6MB | 2.6MB | -496.0B |
| `esql` | 183.9KB | 185.5KB | +1.6KB |
| `eventAnnotation` | 7.2KB | 7.2KB | +26.0B |
| `eventAnnotationListing` | 227.7KB | 229.5KB | +1.9KB |
| `exploratoryView` | 182.7KB | 182.9KB | +162.0B |
| `expressionError` | 14.3KB | 14.3KB | -11.0B |
| `expressionPartitionVis` | 35.5KB | 35.5KB | -1.0B |
| `expressionRevealImage` | 1.6KB | 1.6KB | +1.0B |
| `filesManagement` | 121.4KB | 123.3KB | +1.9KB |
| `fleet` | 1.7MB | 1.7MB | -26.0B |
| `graph` | 414.0KB | 415.9KB | +1.9KB |
| `grokdebugger` | 11.0KB | 11.1KB | +162.0B |
| `home` | 151.9KB | 152.1KB | +173.0B |
| `imageEmbeddable` | 69.0KB | 69.3KB | +324.0B |
| `indexLifecycleManagement` | 166.0KB | 166.1KB | +173.0B |
| `indexManagement` | 693.8KB | 694.7KB | +890.0B |
| `infra` | 1.7MB | 1.7MB | +2.5KB |
| `ingestPipelines` | 405.2KB | 405.3KB | +5.0B |
| `inputControlVis` | 52.1KB | 52.2KB | +162.0B |
| `inventory` | 236.5KB | 236.2KB | -325.0B |
| `investigateApp` | 483.6KB | 481.8KB | -1.8KB |
| `kibanaOverview` | 54.7KB | 54.8KB | +186.0B |
| `kibanaReact` | 197.1KB | 197.1KB | +1.0B |
| `kibanaUtils` | 60.8KB | 60.8KB | -3.0B |
| `lens` | 1.5MB | 1.5MB | +32.0B |
| `licenseManagement` | 46.5KB | 46.7KB | +167.0B |
| `links` | 52.6KB | 52.9KB | +324.0B |
| `logstash` | 33.4KB | 33.6KB | +166.0B |
| `management` | 48.3KB | 48.5KB | +166.0B |
| `maps` | 3.0MB | 3.0MB | +1.8KB |
| `metricsDataAccess` | 99.5KB | 99.6KB | +162.0B |
| `ml` | 4.7MB | 4.7MB | +1.5KB |
| `monitoring` | 533.4KB | 533.6KB | +201.0B |
| `observability` | 471.7KB | 472.5KB | +784.0B |
| `observabilityAIAssistantApp` | 239.2KB | 237.5KB | -1.7KB |
| `observabilityLogsExplorer` | 147.1KB | 147.3KB | +162.0B |
| `observabilityOnboarding` | 271.7KB | 271.4KB | -322.0B |
| `painlessLab` | 18.6KB | 18.8KB | +167.0B |
| `profiling` | 406.0KB | 406.1KB | +167.0B |
| `remoteClusters` | 79.2KB | 79.3KB | +168.0B |
| `rollup` | 115.5KB | 115.7KB | +167.0B |
| `savedObjectsFinder` | 5.2KB | 7.4KB | +2.2KB |
| `savedObjectsManagement` | 84.6KB | 86.8KB | +2.2KB |
| `searchAssistant` | 161.9KB | 160.2KB | -1.7KB |
| `searchHomepage` | 54.8KB | 55.0KB | +162.0B |
| `searchIndices` | 166.7KB | 168.9KB | +2.2KB |
| `searchInferenceEndpoints` | 77.5KB | 77.7KB | +162.0B |
| `searchNotebooks` | 14.2KB | 12.0KB | -2.2KB |
| `searchPlayground` | 167.3KB | 168.6KB | +1.3KB |
| `searchprofiler` | 48.2KB | 48.4KB | +174.0B |
| `securitySolution` | 13.4MB | 13.4MB | -1.6KB |
| `serverlessSearch` | 338.5KB | 340.0KB | +1.5KB |
| `share` | 4.2KB | 3.6KB | -545.0B |
| `slo` | 854.4KB | 865.3KB | +11.0KB |
| `snapshotRestore` | 262.7KB | 262.8KB | +163.0B |
| `synthetics` | 1.1MB | 1.1MB | -805.0B |
| `timelines` | 32.8KB | 32.9KB | +162.0B |
| `transform` | 473.1KB | 473.3KB | +195.0B |
| `triggersActionsUi` | 1.7MB | 1.7MB | +1.1KB |
| `uptime` | 468.1KB | 467.4KB | -736.0B |
| `ux` | 167.6KB | 167.3KB | -335.0B |
| `visualizations` | 316.2KB | 318.2KB | +2.0KB |
| `watcher` | 166.2KB | 166.4KB | +175.0B |
| total |  |  | +42.2KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `@kbn/optimizer` | 10 | 9 | -1 |
| `savedObjects` | 5 | 3 | -2 |
| `savedObjectsFinder` | 0 | 1 | +1 |
| total |  |  | -2 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `advancedSettings` | 7.4KB | 7.6KB | +162.0B |
| `aiops` | 16.1KB | 16.2KB | +70.0B |
| `apm` | 37.9KB | 38.0KB | +70.0B |
| `banners` | 10.4KB | 10.6KB | +162.0B |
| `cases` | 158.2KB | 158.3KB | +162.0B |
| `cloudLinks` | 30.2KB | 30.4KB | +168.0B |
| `console` | 38.3KB | 38.3KB | -23.0B |
| `core` | 454.0KB | 453.9KB | -183.0B |
| `dashboard` | 51.8KB | 52.0KB | +163.0B |
| `dashboardEnhanced` | 17.9KB | 18.0KB | +167.0B |
| `data` | 421.5KB | 421.7KB | +162.0B |
| `dataViewEditor` | 13.5KB | 13.7KB | +162.0B |
| `dataViewFieldEditor` | 26.8KB | 26.9KB | +162.0B |
| `dataVisualizer` | 24.5KB | 24.5KB | +70.0B |
| `embeddable` | 72.1KB | 72.2KB | +162.0B |
| `enterpriseSearch` | 53.8KB | 53.8KB | +70.0B |
| `expressionError` | 7.5KB | 5.2KB | -2.3KB |
| `expressionGauge` | 16.3KB | 16.4KB | +162.0B |
| `expressionHeatmap` | 17.2KB | 17.3KB | +162.0B |
| `expressionImage` | 6.8KB | 4.4KB | -2.4KB |
| `expressionLegacyMetricVis` | 12.5KB | 12.7KB | +162.0B |
| `expressionMetric` | 8.1KB | 5.7KB | -2.4KB |
| `expressionMetricVis` | 16.9KB | 17.1KB | +162.0B |
| `expressionPartitionVis` | 29.1KB | 29.3KB | +163.0B |
| `expressionRepeatImage` | 8.6KB | 6.2KB | -2.3KB |
| `expressionRevealImage` | 8.5KB | 6.3KB | -2.3KB |
| `expressionShape` | 23.1KB | 20.8KB | -2.3KB |
| `expressionTagcloud` | 13.2KB | 13.4KB | +162.0B |
| `expressionXY` | 42.0KB | 42.1KB | +167.0B |
| `fleet` | 173.9KB | 174.1KB | +162.0B |
| `globalSearchBar` | 29.9KB | 30.0KB | +162.0B |
| `guidedOnboarding` | 29.6KB | 29.7KB | +183.0B |
| `indexManagement` | 46.3KB | 46.5KB | +162.0B |
| `ingestPipelines` | 17.5KB | 17.7KB | +238.0B |
| `inspector` | 24.9KB | 25.1KB | +162.0B |
| `interactiveSetup` | 59.6KB | 59.7KB | +169.0B |
| `investigateApp` | 6.7KB | 6.7KB | -19.0B |
| `kbnUiSharedDeps-srcJs` | 3.4MB | 3.4MB | +4.3KB |
| `kibanaReact` | 39.4KB | 39.1KB | -266.0B |
| `kibanaUtils` | 68.7KB | 66.9KB | -1.9KB |
| `lens` | 50.8KB | 51.0KB | +162.0B |
| `licensing` | 11.7KB | 11.9KB | +162.0B |
| `maps` | 54.4KB | 54.5KB | +162.0B |
| `mockIdpPlugin` | 10.6KB | 10.0KB | -601.0B |
| `navigation` | 18.1KB | 18.2KB | +20.0B |
| `newsfeed` | 12.7KB | 12.9KB | +162.0B |
| `observability` | 103.6KB | 103.6KB | -28.0B |
| `observabilityAIAssistantApp` | 8.7KB | 8.7KB | -19.0B |
| `observabilityOnboarding` | 10.4KB | 10.5KB | +70.0B |
| `osquery` | 46.9KB | 47.1KB | +162.0B |
| `presentationPanel` | 43.1KB | 43.4KB | +310.0B |
| `reporting` | 50.7KB | 50.8KB | +162.0B |
| `runtimeFields` | 13.9KB | 14.0KB | +162.0B |
| `savedObjects` | 19.6KB | 17.8KB | -1.9KB |
| `savedObjectsTagging` | 22.6KB | 22.8KB | +162.0B |
| `searchAssistant` | 4.8KB | 4.7KB | -19.0B |
| `searchNotebooks` | 16.4KB | 16.2KB | -124.0B |
| `searchPlayground` | 7.2KB | 7.2KB | -55.0B |
| `security` | 67.8KB | 68.0KB | +162.0B |
| `securitySolution` | 86.2KB | 86.1KB | -72.0B |
| `serverless` | 14.4KB | 14.6KB | +162.0B |
| `share` | 57.6KB | 57.6KB | +7.0B |
| `slo` | 28.6KB | 28.7KB | +77.0B |
| `spaces` | 34.3KB | 34.4KB | +162.0B |
| `synthetics` | 37.5KB | 37.5KB | +70.0B |
| `telemetry` | 24.8KB | 25.0KB | +162.0B |
| `triggersActionsUi` | 127.4KB | 128.8KB | +1.4KB |
| `uiActions` | 23.5KB | 23.6KB | +166.0B |
| `unifiedSearch` | 38.6KB | 38.7KB | +162.0B |
| `upgradeAssistant` | 24.6KB | 24.7KB | +162.0B |
| `uptime` | 22.8KB | 22.9KB | +70.0B |
| `ux` | 6.9KB | 6.9KB | +70.0B |
| `visDefaultEditor` | 24.0KB | 24.1KB | +162.0B |
| `visTypeMarkdown` | 9.7KB | 9.8KB | +162.0B |
| `visTypeTable` | 19.2KB | 19.4KB | +162.0B |
| `visTypeTimelion` | 13.4KB | 13.6KB | +162.0B |
| `visTypeTimeseries` | 20.4KB | 20.6KB | +162.0B |
| `visTypeVega` | 36.1KB | 36.3KB | +167.0B |
| `visTypeVislib` | 15.3KB | 15.5KB | +162.0B |
| `visualizations` | 70.9KB | 71.1KB | +162.0B |
| total |  |  | -4.9KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `@kbn/core-status-common` | 12 | 33 | +21 |
| `@kbn/core-status-common-internal` | 21 | - | -21 |
| `@kbn/core-theme-browser` | 6 | 7 | +1 |
| `@kbn/core-ui-settings-common` | 34 | 48 | +14 |
| `@kbn/index-management-shared-types` | 124 | 126 | +2 |
| `@kbn/management-settings-ids` | 139 | 140 | +1 |
| `@kbn/monaco` | 123 | 124 | +1 |
| `@kbn/react-hooks` | 8 | 9 | +1 |
| `@kbn/react-kibana-context-common` | 8 | 15 | +7 |
| `@kbn/react-kibana-context-theme` | 11 | 13 | +2 |
| `@kbn/shared-ux-table-persist` | 3 | 17 | +14 |
| `@kbn/ui-shared-deps-src` | 59 | 60 | +1 |
| `indexManagement` | 243 | 244 | +1 |
| `savedObjects` | 114 | 97 | -17 |
| `securitySolution` | 188 | 187 | -1 |
| total |  |  | +27 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `slo` | 30 | 31 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/c473a69afc44d61007ba883b03179d28173d0a47) | [after](https://github.com/elastic/kibana/commit/5c485fa728713f6e8fe03193c0bb6aa08be49f14) | diff |
| --- | --- | --- | --- |
| `@kbn/monaco` | 10 | 8 | -2 |
| `savedObjects` | 62 | 11 | -51 |
| `securitySolution` | 448 | 438 | -10 |
| total |  |  | -63 |


</details>


### History
* :broken_heart: [Build #252005](https://buildkite.com/elastic/kibana-pull-request/builds/252005) failed 0584227c0b58a38162155c3f05abc039eb852967
* :broken_heart: [Build #251970](https://buildkite.com/elastic/kibana-pull-request/builds/251970) failed e2393e5304938bb7c055e228f841776a8aa78c3a

cc @jesuswr

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/252111","number":252111,"commit":"5c485fa728713f6e8fe03193c0bb6aa08be49f14"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/252005","number":252005,"commit":"0584227c0b58a38162155c3f05abc039eb852967"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/251970","number":251970,"commit":"e2393e5304938bb7c055e228f841776a8aa78c3a"}],"number":252111}
buildkite-pr-comment-->
